src/query/v2/accessors.hpp

```diff
@@ -140,6 +149,7 @@ class VertexAccessor final {
  private:
   Vertex vertex;
   std::vector<std::pair<PropertyId, Value>> properties;
+  const msgs::ShardRequestManagerInterface *manager_;
```

Comment:
I ended up using the `NameIdMapper` in ShardMap.

Reply:
I think keeping the ShardMap is as simple as possible is a good thing, because the ShardMap structure should be used to communicate the information. However by introducing `NameIdMapper` we are kind of mixing responsibilities of different layers. We already had some not-the-best implementation things (e.g.: the ShardMap should be used to communicate the information only, not to fetch shards/mappings/etc) and with this change we just deepened this mistake.

However it is possible this is only my opinion, so if it is okay for the others, then it is also okay for me.